src/modules/todo.js

```diff
@@ -1,39 +1,138 @@
 import { createElement, $ } from './utils.js';
 
 export default class Todo {
-  constructor(tasks = [], targetContainer = $('ul')) {
-    this.tasks = tasks;
+  LOCAL_STORAGE_KEY = 'tasks';
+
+  constructor(targetContainer = $('ul'), form = $('form')) {
+    this.tasks = this.getFromLocalStorage();
     this.targetContainer = targetContainer;
+    form.addEventListener('submit', (e) => this.handleSubmit(e));
+    this.editIndex = -1;
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { todo } = event.target;
+    this.addTask(todo.value);
+    event.target.reset();
   }
 
   addTask(task) {
-    this.tasks.push(task);
+    this.tasks.push({
+      description: task,
+      index: this.tasks.length,
+      completed: false,
+    });
+    this.saveToLocalStorage();
+    this.render();
+  }
+
+  removeTask(index) {
+    this.tasks = this.tasks.reduce((prev, curr) => {
+      if (curr.index !== index) {
+        curr.index = prev.length;
+        prev.push(curr);
+      }
+      return prev;
+    }, []);
+    this.closeEdit();
```

Comment:
- [ ] After deleting a task here, it is not removed from the local storage. Please make sure that after a user deletes a task, it is also removed from the local storage 👍

Reply:
> Oh, Ok, Thank you @Lucas-Masaba, I didn't noticed that, I will fix it